xbmc/utils/LangCodeExpander.h

```diff
@@ -102,22 +102,53 @@ class CLangCodeExpander
   */
   std::string ConvertToISO6392T(const std::string& lang);
 
+  /*
+   * \brief Find a language code with subtag (e.g. zh-tw, zh-Hans) in to a string.
+   *        This function find a limited set of IETF BCP47 specs, so:
+   *        language tag + region subtag, or, language tag + script subtag.
+   *        The language code can be found also if wrapped with round brackets.
+   * \param str [IN] The string where find the language code.
+   * \param langCode [OUT] The language code found in the string.
+   * \return True if language code found, otherwise false.
+   */
+  static bool FindLanguageCodeWithSubtag(const std::string& str, std::string& langCode);
+
 #ifdef TARGET_WINDOWS
   static bool ConvertISO31661Alpha2ToISO31661Alpha3(const std::string& strISO31661Alpha2, std::string& strISO31661Alpha3);
   static bool ConvertWindowsLanguageCodeToISO6392B(const std::string& strWindowsLanguageCode, std::string& strISO6392B);
 #endif
 
-  std::vector<std::string> GetLanguageNames(LANGFORMATS format = ISO_639_1, bool customNames = false);
-protected:
+  /*
+   * \brief Get the list of language names.
+   * \param format [OPT] The format type.
+   * \param addAddonLanguages [OPT] True to add languages of addons languages.
+   * \param addUserDefined [OPT] True to add user-defined languages (can override existings).
+   * \return The languages
+   */
+  std::vector<std::string> GetLanguageNames(LANGFORMATS format = ISO_639_1,
```

Comment:
Can you change both bools to an enum class? Would make the calling code easier to read (rather than having true,true,false,false, etc)

Reply:
done, but LANGFORMATS enum seem cannot be changed to class enum without doing a breaking change on python interface so i kept it as is